src/stacked_borrows.rs

```diff
@@ -240,17 +240,13 @@ impl<'tcx> Stack {
                 granting + 1,
             Permission::SharedReadWrite => {
                 // The SharedReadWrite *just* above us are compatible, to skip those.
-                let mut idx = granting + 1;
-                while let Some(item) = self.borrows.get(idx) {
-                    if item.perm == Permission::SharedReadWrite {
-                        // Go on.
-                        idx += 1;
-                    } else {
+                for (idx, item) in self.borrows.iter().enumerate().skip(granting + 1) {
+                    if item.perm != Permission::SharedReadWrite {
                         // Found first incompatible!
-                        break;
+                        return idx;
                     }
                 }
-                idx
+                self.borrows.len()
```

Comment:
It is this redundancy that I did not use an iterator before -- TBH the mutable version seemed cleaner to me, if the alternative *still* involved explicitly querying the vector's length.

I won't fight over this though if you really want it changed.^^

Reply:
yea it's not ideal, I'll revert it. What about the two changes below?

Reply:
*shrug* no strong preference